packages/apollo-voyager-keycloak/src/AuthContextProvider.ts

```diff
@@ -1,31 +1,27 @@
 import { Request } from 'express'
 import { IncomingMessage } from 'http'
-
-export class KeycloakAuthContextProvider {
-
+import { AuthContextProvider } from '@aerogear/apollo-voyager-server';
+export class KeycloakAuthContextProvider implements AuthContextProvider {
   public readonly request: any
   public readonly accessToken: any
   public readonly authenticated: boolean
 
-  constructor (request: any) {
-    this.request = request
-    this.accessToken = (request && request.kauth && request.kauth.grant) ? request.kauth.grant.access_token : undefined
-    this.authenticated = !!(this.accessToken)
-  }
-
-  public getToken () {
-    return this.accessToken
+  constructor ({ req }: { req: any }) {
+    this.request = req
+    this.accessToken = (req && req.kauth && req.kauth.grant) ? req.kauth.grant.access_token : undefined
+    this.authenticated = !!((req && req.kauth && req.kauth.grant) ? req.kauth.grant.access_token : undefined)
   }
 
   public isAuthenticated () {
     return this.authenticated
   }
 
-  public getTokenContent () {
-    return this.isAuthenticated() ? this.getToken().content : null
+  public hasRole(role: string): boolean {
+    return this.isAuthenticated() && this.accessToken.hasRole(role)
   }
 
-  public hasRole (role: string) {
-    return this.isAuthenticated() && this.getToken().hasRole(role)
+  // TODO where can we get this from?? The accessToken is kind of like a user but not exactly.
+  public getUser() {
+    return null
```

Comment:
@wtrocki any ideas about this one?

Reply:
Checked and it's not possible to do it. Users will be able to get email from kauth and then perform queries, but rest of the data is missing. There is separate endpoint to get all principal data, but technically application itself can have access to their own user storage.

Reply:
@wtrocki This is kind of what I was thinking. Should I bring back `getToken()` for the time being?